src/studio/src/designer/backend/Services/Implementation/ConvertTextsService.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using Altinn.Studio.Designer.Models;
+using Altinn.Studio.Designer.Services.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using RestSharp;
+
+namespace Altinn.Studio.Designer.Services.Implementation
+{
+    /// <summary>
+    /// Service for converting texts files into new format.
+    /// </summary>
+    public class ConvertTextsService : IConvertTextsService
+    {
+        private readonly IAltinnGitRepositoryFactory _altinnGitRepositoryFactory;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="altinnGitRepositoryFactory">IAltinnGitRepository</param>
+        public ConvertTextsService(IAltinnGitRepositoryFactory altinnGitRepositoryFactory)
+        {
+            _altinnGitRepositoryFactory = altinnGitRepositoryFactory;
+        }
+
+        /// <inheritdoc />
+        public async void ConvertV1TextsToV2(string org, string repo, string developer)
+        {
+            var altinnAppGitRepository = _altinnGitRepositoryFactory.GetAltinnAppGitRepository(org, repo, developer);
+
+            var languageFiles = altinnAppGitRepository.FindFiles(new string[] { "resource.*.json" });
+
+            foreach (string languageFile in languageFiles.ToList())
+            {
+                Dictionary<string, string> newTexts = new Dictionary<string, string>();
+
+                string fileName = Path.GetFileName(languageFile);
+                string languageCode = fileName.Split(".")[1];
```

Comment:
For readability consider creating a small method `GetLanguageCodeFromFileName` - this reveals the intention and you don't have to think or guess it byt the usage.